source/common/upstream/outlier_detection_impl.h

```diff
@@ -39,16 +39,34 @@ class DetectorImplFactory {
 
 class DetectorImpl;
 
+class SRAccumulatorImpl {
+public:
+  SRAccumulatorImpl()
+      : current_sr_bucket_(new SRAccumulatorBucket()),
+        backup_sr_bucket_(new SRAccumulatorBucket()){};
+  SRAccumulatorBucket* getCurrentWriter();
+  Optional<double> getSR(uint64_t rq_volume_threshold);
```

Comment:
I feel that I have used SR everywhere, and expanding it here (while useful, and does not make the name too long) would be confusing.